api/config/v1alpha1/envoygateway_types.go

```diff
@@ -135,6 +141,76 @@ type RateLimitRedisSettings struct {
 	URL string `json:"url"`
 }
 
+// Extension defines the configuration for registering an extension to
+// the Envoy Gateway control plane.
+type Extension struct {
+	// Resources defines the set of K8s resources the extension will handle.
+	//
+	// +optional
+	Resources []GroupVersionKind `json:"resources,omitempty"`
+
+	// Hooks defines the set of hooks the extension supports
+	//
+	// +kubebuilder:validation:Required
+	Hooks *ExtensionHooks `json:"hooks,omitempty"`
+
+	// Service defines the configuration of the extension service that the Envoy
+	// Gateway Control Plane will call through extension hooks.
+	//
+	// +kubebuilder:validation:Required
+	Service *ExtensionService `json:"service,omitempty"`
+}
+
+// ExtensionHooks defines extension hooks across all supported runners
+type ExtensionHooks struct {
+	// XDS defines all the supported extension hooks for the XDS runner
+	XDS *XDSHooks `json:"xds,omitempty"`
```

Comment:
is this tied to the [xds-server](https://github.com/envoyproxy/gateway/blob/main/internal/xds/server/runner/runner.go) or [xds-translator](https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/runner/runner.go)

Reply:
XDS translator. I'll update to make it more clear